### PR TITLE
Add weather forecast widget to demonstrate:

### PR DESCRIPTION
**Pull request by thagikura**
- Responsive size layouts
- Dynamic colors
- Rounded corners including outer container and inner views
- Improved preview picker experience

Screenshots are as follows for each size and light/dark theme.

weather_widget_light_small

weather_widget_light_medium

weather_widget_light_large

weather_widget_dark_small

weather_widget_dark_medium

weather_widget_dark_large

### CODE REVIEW

**Comment by thagikura** [comment]
@marcelpinto FYI. I'd appreciate you to have a look.

File: AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    private fun updateWeatherWidget(
+        context: Context,
+        appWidgetManager: AppWidgetManager,
+        appWidgetId: Int
+    ) {
+        val viewMapping: MutableMap<SizeF, RemoteViews> = mutableMapOf()
+        // Specify the maximum width and height in dp and a layout, which you want to use for the
+        // specified size
```
Comment (yaraki) [inline AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt:48]: I couldn't understand what this comment means. Does the system choose one of these instances of RemoteViews depending on the available area?

Comment (thagikura) [reply AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt:48]: The system picks up the layout depending on the widget size being resized.
Added more explanation about that.

Comment (yaraki) [reply AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt:48]: The comment says "maxium width and height", but the explanation below seems to say that they are minimum size?

**Review by marcelpinto — APPROVED** [review]
LGTM just two minor nits

File: AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt (hunk 1 above)
Comment (thagikura) [reply AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt]: Oh that's right. Fixed the comment, thanks.

File: AppWidget/app/src/main/res/layout/widget_weather_forecast_large.xml [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            android:minHeight="@dimen/widget_element_min_length" >
+
+            <TextView
+                style="@style/TextAppearance.AppCompat.Large"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="end"
+                android:gravity="bottom"
+                android:text="@string/city_tokyo"
+                android:textColor="?android:attr/textColorPrimary" />
```
[the remark this answers is not in the document]
Comment (thagikura) [reply AppWidget/app/src/main/res/layout/widget_weather_forecast_large.xml]: Done. Extracted the textColor to its theme.

File: AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        // In the following case:
+        //   - R.layout.widget_weather_forecast_small is used from
+        //     180dp (or minResizeWidth) x 110dp (or minResizeHeight) to 269dp (next cutoff point - 1) x 279dp (next cutoff point - 1)
+        //   - R.layout.widget_weather_forecast_medium is used from 270dp x 110dp to 270dp x 279dp (next cutoff point - 1)
+        //   - R.layout.widget_weather_forecast_large is used from
+        //     270dp x 280dp to 570dp (specified as maxResizeWidth) x 450dp (specified as maxResizeHeight)
+        viewMapping[SizeF(180.0f, 110.0f)] = RemoteViews(
+                        context.packageName,
```
[the remark this answers is not in the document]
Comment (thagikura) [reply AppWidget/app/src/main/java/com/example/android/appwidget/WeatherForecastAppWidget.kt]: Done.

**Review by yaraki — APPROVED** [review]

File: AppWidget/app/src/main/res/layout/widget_weather_forecast_large.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sakibguy) [reply AppWidget/app/src/main/res/layout/widget_weather_forecast_large.xml]: ok

**Pull request merged**